cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -919,17 +919,13 @@ private V1PodSpec createPodSpec(Location runtimeConfigLocation, Map<String, Runt
     volumeMounts.add(new V1VolumeMount().name("workdir").mountPath(workDir));
     volumeMounts.addAll(Arrays.asList(extraMounts));
 
-    // Add secret disks as secret volume mounts
-    List<V1Volume> secretVolumes = new ArrayList<>();
-    if (secretDiskRunnables.containsKey(runnableName)) {
-      for (SecretDisk secretDisk : secretDiskRunnables.get(runnableName).getSecretDisks()) {
-        String secretName = secretDisk.getName();
-        String mountPath = secretDisk.getPath();
-        secretVolumes.add(new V1Volume().name(secretName).secret(new V1SecretVolumeSource()
-                                                                   .secretName(secretName)));
-        volumeMounts.add(new V1VolumeMount().name(secretName).mountPath(mountPath).readOnly(true));
-      }
-    }
+    // Mount all volumes including cdap-secret for init container as it runs system/trusted code.
+    List<V1VolumeMount> initContainerVolumeMounts = new ArrayList<>(volumeMounts);
```

Comment:
nit: I assume we don't need secrets for downloading artifacts from GCS in the init container. Right?

Reply:
It has already been changed to download via appfab